hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -239,12 +240,11 @@ public Comparable getOrderingValue(Option<T> recordOption,
     if (metadataMap.containsKey(INTERNAL_META_ORDERING_FIELD)) {
       return (Comparable) metadataMap.get(INTERNAL_META_ORDERING_FIELD);
     }
-
     if (!recordOption.isPresent() || !orderingFieldTypeOpt.isPresent()) {
       return orderingFieldDefault;
     }
-
-    Object value = getValue(recordOption.get(), schema, orderingFieldName);
+    Schema schemaApplied = metadataMap.containsKey(INTERNAL_META_SCHEMA) ? (Schema) metadataMap.get(INTERNAL_META_SCHEMA) : schema;
+    Object value = getValue(recordOption.get(), schemaApplied, orderingFieldName);
```

Comment:
Does this fix a bug or the schema passed in is always the same as `metadataMap.get(INTERNAL_META_SCHEMA)`?

Reply:
I did see some failures because of incorrect schema. So I try to apply the internal meta schema first. May be it is temporary. I can remove it to see if any tests fail.